pkg/messaging/topics.go

```diff
@@ -154,3 +137,70 @@ func EncodeMessageMQTTTopic(m *Message) string {
 	}
 	return topic
 }
+
+// ParseTopic parses a messaging topic string and returns the domain ID, channel ID, and subtopic.
+// This is an optimized version with no regex and minimal allocations.
+func ParseTopic(topic string) (domainID, chanID, subtopic string, err error) {
+	// location of string "m"
+	start := 0
+	// Handle both formats: "/m/domain/c/channel/subtopic" and "m/domain/c/channel/subtopic".
+	// If topic start with m/ then start is 0 , If topic start with /m/ then start is 1.
+	n := len(topic)
+	if n > 0 && topic[0] == '/' {
+		start = 1
+	}
+
+	// length check - minimum: "m/<domain_id>/c/" = 5 characters if ignore <domain_id> and in this case start will be 0
+	// length check - minimum: "/m/<domain_id>/c/" = 6 characters if ignore <domain_id> and in this case start will be 1
+	if n < start+5 {
+		return "", "", "", ErrMalformedTopic
+	}
+	if topic[start] != 'm' || topic[start+1] != '/' {
+		return "", "", "", ErrMalformedTopic
+	}
+	pos := start + 2
+
+	// Find "/c/" to locate domain ID
+	cPos := -1
+	for i := pos; i <= n-3; i++ {
+		if topic[i] == '/' && topic[i+1] == 'c' && topic[i+2] == '/' {
```

Comment:
We can use constant for Prefix chars like below

```suggestion
		if topic[i] == '/' && topic[i+1] == ChannelTopicPrefix && topic[i+2] == '/' {
```
But this requires following changes 

```go 
const SubjectAllMessages = string(messaging.MsgTopicPrefix) + ".>"

const (
	MsgTopicPrefix     = 'm'
	ChannelTopicPrefix = 'c'
)

```